transports/vm/src/test/java/org/mule/transport/vm/VMQueueTestCase.java

```diff
@@ -95,6 +104,28 @@ public void testPassThrough() throws Exception
         }
     }
 
+    @Test
+    public void testPassThroughKeepsMimeType() throws Exception
+    {
+        MuleClient client = muleContext.getClient();
+        Set<String> polos = new HashSet<String>(Arrays.asList(new String[]{"Marco", "Niccolo", "Maffeo"}));
+        Iterator<String> people = polos.iterator();
+        SimpleDataType<Object> customDataType = new SimpleDataType<>(String.class, "application/custom");
+        while (people.hasNext())
+        {
+            DefaultMuleMessage message = new DefaultMuleMessage(people.next(), null, (Map<String, Object>) null, null, muleContext, customDataType);
+            client.dispatch("vm://entry", message);
+        }
+        
+        for (int i = 0; i < 3; ++i)
```

Comment:
Maybe a loop through the collection (or at least a for with the collection size/array length as limit).